### PR TITLE
update platform linux68k

### PR DESCRIPTION
**Pull request by tevorbl**
latest version of musashi engine
includes floating point emulation
(plus a few patches to add in missing opcodes needed by ack - see tags JFF & TBB)

added a few missing linux syscalls in sim.c

pascal now runs pretty well
quick test with modula2 passes

c gets the floating point numbers wrong, so more work needed here.  Basic hello world type programs seem OK, though.

other languages untested

plat/linux68k/emu/build.lua is probably not quite right - the softfloat directory is compiled in the wrong place.   Any help here would be appreciated.

### CODE REVIEW

File: plat/linux68k/emu/sim.c [hunk 1]
```diff
@@ -310,11 +361,15 @@ int main(int argc, char* argv[])
 
 	load_program(fhandle);
 
-//	disassemble_program();
+        //disassemble_program();
 
-	m68k_set_cpu_type(M68K_CPU_TYPE_68020);
-	m68k_init();
-	m68k_pulse_reset();
+        //printf("now at line %d\n", __LINE__);
```
Comment (davidgiven) [inline plat/linux68k/emu/sim.c]: stray tracing

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,8 @@ DEFAULT_PLATFORM = pc86
 
 # Where should the ACK put its temporary files?
 
-ACK_TEMP_DIR = /tmp
+#ACK_TEMP_DIR = /tmp
+ACK_TEMP_DIR = tmp
```
Comment (davidgiven) [inline Makefile]: please revert this file

**Comment by davidgiven** [comment]
Thank you very much for this! Particularly as I was halfway through explaining how I couldn't use it because new the license had a non-commercial use clause before realising I had the patch backwards and the _old_ version had the non-commercial use clause. Er, yikes. So, er, definitely thanks.

I reckon it's pretty much good to go, apart from a couple of really nitpicky comments about stray comments and debugging changes.

(I actually have another compiler project which could really use this, too...)

File: plat/linux68k/emu/sim.c (hunk 1 above)
Comment (tevorbl) [reply plat/linux68k/emu/sim.c]: OK, should be done now.
formatting is leading tabs, stray tracing removed.
Also restored temp dir & CC in top level Makefile

**Review by davidgiven — APPROVED** [review]

**Comment by davidgiven** [comment]
Fantastic, thanks! Merging.

**Pull request merged**